app/models/vm_question_response.rb

```diff
@@ -0,0 +1,244 @@
+#represents each table in the view_team view.
+#the important piece to note is that the @listofrows is a  list of type VmQuestionResponse_Row, which represents a row of the heatgrid table.
+class VmQuestionResponse
+
+  def initialize(questionnaire, round,rounds)
+      @listofrows = []
+      @listofreviewers = []
+      @listofreviews = []
+      @listofteamparticipants = []
+      @max_score = questionnaire.max_question_score
+      @questionnaire_type = questionnaire.type
+      @questionnaire_display_type = questionnaire.display_type
+      @rounds = rounds
+      @round = round
+    @name  = questionnaire.name
+  end
+
+  def name
+    @name
+  end
+
+  def addQuestions(questions)
+    questions.each do |question|
+      # Get the maximum score for this question. For some unknown, godforsaken reason, the max
+      # score for the question is stored not on the question, but on the questionnaire. Neat.
+      corresponding_questionnaire = Questionnaire.find_by(id: question.questionnaire.id)
+      question_max_score = corresponding_questionnaire.max_question_score
+      row = VmQuestionResponseScoreRow.new(question.txt, question.id, question.weight, question_max_score,question.seq)
+      @listofrows << row
+    end
+  end
+
+  def addReviews(participant,team,vary)
+
+    if @questionnaire_type == "ReviewQuestionnaire"
+      if @rounds == 1
+        reviews = participant.reviews()
+      else
+        reviews = ResponseMap.get_assessments_for_round(team ,@round,@rounds)
```

Comment:
Yes, those two 'view_reviewer' are duplicated.
I'm not sure, I'll forward this email to my teammates.

2015-12-14 11:15 GMT-05:00 Yang Song <notifications@github.com>:

> In app/models/vm_question_response.rb
> <https://github.com/expertiza/expertiza/pull/636#discussion_r47517179>:
>
> > +      # Get the maximum score for this question. For some unknown, godforsaken reason, the max
> > +      # score for the question is stored not on the question, but on the questionnaire. Neat.
> > +      corresponding_questionnaire = Questionnaire.find_by(id: question.questionnaire.id)
> > +      question_max_score = corresponding_questionnaire.max_question_score
> > +      row = VmQuestionResponseScoreRow.new(question.txt, question.id, question.weight, question_max_score,question.seq)
> > +      @listofrows << row
> > +    end
> > +  end
> > +
> > +  def addReviews(participant,team,vary)
> > +
> > +    if @questionnaire_type == "ReviewQuestionnaire"
> > +      if @rounds == 1
> > +        reviews = participant.reviews()
> > +      else
> > +        reviews = ResponseMap.get_assessments_for_round(team ,@round,@rounds)
>
> What are round and round? Those names are confusing to me.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/expertiza/expertiza/pull/636/files#r47517179>.
>